common/unit_test/Test_Common_Iota.hpp

```diff
@@ -0,0 +1,105 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef TEST_COMMON_IOTA_HPP
+#define TEST_COMMON_IOTA_HPP
+
+#include <type_traits>
+
+#include "KokkosKernels_Iota.hpp"
+
+template <typename T, typename Size>
+void test_iota_constructor() {
+  // empty iota
+  {
+    Iota<T, Size> i;
+    EXPECT_EQ(i.size(), 0);
+  }
+
+  // basic iota
+  {
+    Iota<T, Size> ten(10);
+    EXPECT_EQ(ten.size(), 10);
+    for (size_t i = 0; i < ten.size(); ++i) {
+      EXPECT_EQ(ten(i), i);
+    }
+  }
+
+  // iota with offset
+  if constexpr (std::is_signed_v<T>) {
+    Iota<T, Size> three(3, -7);
+    EXPECT_EQ(three.size(), 3);
+    for (size_t i = 0; i < three.size(); ++i) {
+      EXPECT_EQ(three(i), T(i) - T(7));
+    }
+  }
+
+  // iota with negative offset
```

Comment:
With positive offset? I'm not sure what the convention for positive/negative offset here?